### PR TITLE
SecRuleEngine parameter for modsecurity

### PR DESCRIPTION
**Pull request by unixninja**
Added a parameter that sets mod security SecRuleEngine. Allowing for DetectionOnly or off.

### CODE REVIEW

File: templates/mod/security.conf.erb [hunk 1]
```diff
@@ -9,7 +9,7 @@
 <%- end -%>
 
     # Default recommended configuration
-    SecRuleEngine On
+    SecRuleEngine <%= @modsec_secruleengine %>
```
Comment (igalic) [inline templates/mod/security.conf.erb]: maybe use our [bool2httpd](https://github.com/puppetlabs/puppetlabs-apache/search?utf8=%E2%9C%93&q=bool2httpd) function here?

**Comment by igalic** [comment]
@unixninja would you mind [squashing](http://www.git-scm.com/book/en/Git-Tools-Rewriting-History#Squashing-Commits) those commits down to one?
****
Also, when you add your email from those two commits to your github profile, this pull request will be counted in your Magick Internet Points.

**Comment by mhaskel** [comment]
@unixninja thanks for the contribution! Any way we could get that parameter renamed to `sec_rule_engine`? We also need updates in the mod::sec [spec test](https://github.com/puppetlabs/puppetlabs-apache/blob/master/spec/classes/mod/security_spec.rb)

**Comment by jonnytpuppet** [comment]
ping @unixninja

**Comment by di-sysops** [comment]
++ to this PR

**Comment by jackdpeterson** [comment]
++ to this Pull Request. This will help in adoption of mod_security by allowing testing out rules prior to pushing the engine to, "On".

**Comment by igalic** [comment]
closing in favour of  #1168

**Pull request closed without merging**